shotover-proxy/src/message/mod.rs

```diff
@@ -74,6 +77,18 @@ impl Message {
                 message_type,
             }),
             meta_timestamp: None,
+            compression: None,
+        }
+    }
+
+    pub fn from_bytes_with_compression(bytes: Bytes, compression: Compression) -> Self {
```

Comment:
We should always be using this for cassandra message creation, otherwise we might forget to include the compression, so can we name it `from_cassandra_bytes`?

Reply:
Maybe we should make `MessageType::Cassandra` into `MessageType::Cassandra { compression: Compression }`.
That way we can just do away with this method entirely.
Semantically that makes a lot of sense since each compression type is effectively its own message type as you cant decode a compressed cassandra message without the knowledge of what compression type was used.
But maybe we'll likely need to fork the MessageType enum as there are probably uses of this enum that dont care about the compression type.


This would also help cleanup Frame as well!

Reply:
I had initially done that but methods like `get_type` and `into_encodable` become messy.

Reply:
Can get_type and into_encodable have their own separate MessageType enum? Not sure what you would call it, maybe get_type/into_encodable have MessageType and from_bytes has a ProtocolType? or maybe a MessageTypeFull?